lib/wasi/src/fs/inode_guard.rs

```diff
@@ -0,0 +1,753 @@
+use std::{
```

Comment:
What's the use of the inode_guard? This seems quite complex, with listeners and events.

Reply:
Its used by the poll syscall which needs to hold multiple locks on multiple files in a safe and performant way. Yes its complex but thats because `poll_oneoff` is complex which has been made considerably simplier and more stable then it was. Ultimately these guards and the pill create an async bridge back to the threads in the runtime using an wasm version of mio and tokio. Not fun to implement but definately worth it as we get some impressive async performance now in wasix

Reply:
Performance by the way which is hard to beat in other runtimes. @syrusakbary we should definately benchmark this and put it on the website

Reply:
Treating this as resolved , since it's just a question - @ptitSeb